src/Context/StateProvider.js

```diff
@@ -0,0 +1,21 @@
+import React from 'react'
+import { createContext } from 'react'
+import { useReducer } from 'react'
+import reducerFunction from '../reducerFunction'
+export const StateContext = createContext()
+
+const StateProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(reducerFunction, {
+    showColorPalette: false,
```

Comment:
Nice way of handling showing/ hiding the color palette or the text editor with the flag!